src/components/tables/DataTable/table/DataTable.tsx

```diff
@@ -18,6 +19,167 @@ import type { DataTableStatus } from '../DataTableContext.tsx';
 
 import cl from './DataTable.module.scss';
 
+type ColElement = React.ReactElement<{ style?: React.CSSProperties }>;
+type IdentifierColumnConfig = { columnId: string, sticky?: boolean };
+type ActionColumnConfig = { columnId: string, sticky?: boolean };
+
+const getFlexValueFromColumn = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col?.bkColumnWidth?.flex ?? 1); // defaults to 1
+};
+
+const getFlexColumnWidth = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col?.bkColumnWidth?.width ?? 0);
+};
+
+const isFlexColumn = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col && Object.hasOwn(col, 'bkColumnWidth'));
+};
+
+const requireNewCol = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (isFlexColumn(col) && (col.bkColumnWidth?.flex ?? 0) > 0);
+};
+
+const calculateFlexPercentage = <D extends object>(
+  allColumns: ReactTable.ColumnInstance<D>[],
+  targetColumn: ReactTable.ColumnInstance<D>
+) => {
+  if (!isFlexColumn(targetColumn)) {
+    return 0;    
+  }
+
+  const targetColumnFlex = getFlexValueFromColumn(targetColumn);
+  const totalFlex = allColumns.reduce((total, column) => total + getFlexValueFromColumn(column), 0);
+  
+  if (targetColumnFlex === 0) {
+    return 0;
+  }
+  
+  return totalFlex > 0 ? Math.round((targetColumnFlex / totalFlex) * 100) : 0;
+};
+
+export function useCustomColGroup<D extends object>(
```

Comment:
Why is this a hook? I don't see any hooks in this function. Also please convert it to an arrow function